File: Tp1/PalindromoRecursivo.cpp
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

void Palindromo(char in[],int i,bool ehPalin){
	if(i < (strlen(in)-1)/2){
		if(in[i] != in[strlen(in)-2-i]){
			ehPalin = false;
			i = strlen(in);
		}
		i++;
		if(ehPalin)
			printf("SIM\n");
		else
			printf("NAO\n");
	}
}

int main(){
	char in[500];
	fgets(in,500,stdin);
	while(!(strlen(in) == 4 && in[0] == 'F' && in[1] == 'I' && in[2] == 'M')){
		Palindromo(in,0,true);
		fgets(in,500,stdin);
	}
	return 0;
}

File: Tp1/PalindromoRecursivo.java
class PalindromoRecursivo{
	
	public static boolean Palindromo(String in,int i,boolean ehPalin){
		if(i < in.length()/2){
			if(in.charAt(i) != in.charAt(in.length()-1-i)){
				ehPalin = false;
				i = in.length();
			}
			i++;
			Palindromo(in,i,ehPalin);
		}
		return ehPalin;
	}

	public static void main(String[] args){
		String in = MyIO.readLine();
		boolean ehPalin;
		while(!(in.length() == 3 && in.charAt(0) == 'F' && in.charAt(1) == 'I' && in.charAt(2) == 'M')){
			ehPalin = true;
			ehPalin = Palindromo(in,0,ehPalin);
			if(ehPalin)
				MyIO.println("SIM");
			else
				MyIO.println("NAO");
			in = MyIO.readLine();
		}
	}
}
